crates/flycheck/src/lib.rs

```diff
@@ -54,7 +54,7 @@ impl fmt::Display for FlycheckConfig {
 pub struct FlycheckHandle {
     // XXX: drop order is significant
     sender: Sender<Restart>,
-    thread: jod_thread::JoinHandle,
+    thread: jod_thread::JoinHandle<()>,
```

Comment:
```
   Compiling flycheck v0.0.0 (/Users/pksunkara/Coding/pksunkara/rust-analyzer/crates/flycheck)
error[E0107]: wrong number of type arguments: expected 1, found 0
  --> /Users/pksunkara/Coding/pksunkara/rust-analyzer/crates/flycheck/src/lib.rs:57:13
   |
57 |     thread: jod_thread::JoinHandle,
   |             ^^^^^^^^^^^^^^^^^^^^^^ expected 1 type argument

error: aborting due to previous error

For more information about this error, try `rustc --explain E0107`.
error: could not compile `flycheck`.
```

I was getting these issues when compiling the crates as path dependencies of another lib. Not sure why `cargo test` isn't failing.

Reply:
>I was getting these issues when compiling the crates as path dependencies of another lib. Not sure why cargo test isn't failing.

You have outdated Cargo.lock in that context.

Reply:
That is, let's stick to new jod_thread here, and remove `()`

Reply:
Done